syz-ci/manager.go

```diff
@@ -910,12 +913,35 @@ func (mgr *Manager) uploadCoverStat(fuzzingMinutes int) error {
 	go func() {
 		decoder := json.NewDecoder(resp.Body)
 		for decoder.More() {
-			var covInfo cover.CoverageInfo
-			if err := decoder.Decode(&covInfo); err != nil {
-				pw.CloseWithError(fmt.Errorf("failed to decode CoverageInfo: %w", err))
+			if err := f(pw, decoder, curTime); err != nil {
```

Comment:
Nit: you could make the `uploadCoverJSONLToGCS` function parameterized and pass down not the `decoder` object, but the the already decoded value.

Reply:
I like the idea... but we can't generalize methods yet. https://github.com/golang/go/issues/49085